### DISCUSSION TITLE
energy sensitive to grid

### DISCUSSION THREAD

**Discussion by tiejundong** [Electrostatics/apbs#235] [Q&A] [answered]
Dear developors:
I want to get a stable energy solution of a small protein in water, but I found that the output is VERY sensitive to the grid even when grid changed from 0.20 to 0.15 (input parameters are listed below). It's unreasonable that energy changes with such a huge gap. And there were no "warning" found in output. I further tested the same input parameters with differently rotated protein, and the outputs were also shew a huge gap.

```
# input_1 with grid 0.15:
read
    mol pqr ./temp/d3cz1a_.pqr
end
elec name solvated
    mg-manual
    dime 481 481 481
    grid 0.15 0.15 0.15
    gcent mol 1
    cgcent mol 1
    mol 1
    npbe
    bcfl mdh
    pdie 2.0000
    sdie 78.5400
    srfm spl4
    chgm spl4
    sdens 10.00
    srad 1.40
    swin 0.30
    temp 298.15
    calcenergy comps
    calcforce no
    write atompot flat ./temp/d3cz1a_
end
print elecEnergy solvated end
quit

# part of the output of input_1:
      Atom 1754:  1.559426252303E+03 kJ/mol
      Atom 1755:  1.770242862830E+02 kJ/mol
      Atom 1756:  3.945275338317E+01 kJ/mol
      Atom 1757:  1.358541020093E+01 kJ/mol
      Atom 1758:  1.588899031749E+00 kJ/mol
      Atom 1759:  1.824025816646E+00 kJ/mol
      Atom 1760:  1.857655847231E+01 kJ/mol
      Atom 1761:  1.906724510673E+01 kJ/mol
      Atom 1762:  1.872793144695E+01 kJ/mol
      Atom 1763:  9.313815024713E-01 kJ/mol
      Atom 1764:  4.905978396383E-01 kJ/mol
      Atom 1765:  5.799347289324E-01 kJ/mol
  Calculating forces...
  Writing atom potentials to ./temp/d3cz1a_.txt
----------------------------------------
PRINT STATEMENTS

print energy 1 (solvated) end
  Local net energy (PE 0) = 4.827093121362E+05 kJ/mol
  Global net ELEC energy = 4.827093121362E+05 kJ/mol
```

```
# input_2 with grid 0.20:
read
    mol pqr ./temp/d3cz1a_.pqr
end
elec name solvated
    mg-manual
    dime 481 481 481
    grid 0.20 0.20 0.20
    gcent mol 1
    cgcent mol 1
    mol 1
    npbe
    bcfl mdh
    pdie 2.0000
    sdie 78.5400
    srfm spl4
    chgm spl4
    sdens 10.00
    srad 1.40
    swin 0.30
    temp 298.15
    calcenergy comps
    calcforce no
    write atompot flat ./temp/d3cz1a_
end
print elecEnergy solvated end
quit

# part of the output of input_2:
      Atom 1754:  1.220220358504E+03 kJ/mol
      Atom 1755:  1.135450748812E+02 kJ/mol
      Atom 1756:  2.571014095769E+01 kJ/mol
      Atom 1757:  9.762744573482E+00 kJ/mol
      Atom 1758:  9.517268151493E-01 kJ/mol
      Atom 1759:  1.488003789074E+00 kJ/mol
      Atom 1760:  1.133899726322E+01 kJ/mol
      Atom 1761:  1.122201790169E+01 kJ/mol
      Atom 1762:  1.103052388191E+01 kJ/mol
      Atom 1763:  6.115196741168E-01 kJ/mol
      Atom 1764:  2.347072623775E-01 kJ/mol
      Atom 1765:  2.718866171497E-01 kJ/mol
  Calculating forces...
  Writing atom potentials to ./temp/d3cz1a_.txt
----------------------------------------
PRINT STATEMENTS

print energy 1 (solvated) end
  Local net energy (PE 0) = 3.419608556637E+05 kJ/mol
  Global net ELEC energy = 3.419608556637E+05 kJ/mol
```

Strangely, when I calculated the polar solvation of it, i.e. output of ```sdie 78.5400``` - output of ```sdie 1.0```. The outputs are very stable with different grid and dime. 
I think I probably missed some important details of APBS or PB. Could some one help this poor postgraduate student.

The input PDB file:
[d3cz1a_.txt](https://github.com/Electrostatics/apbs/files/8666520/d3cz1a_.txt)

**Comment by sobolevnrm** [comment, marked as answer]
Hello --

This is the expected behavior.

The single-point energy you list in the first example includes "self energies":  these are the energies of the discretized charge distributions interacting with themselves.  The width of these discretized charge distributions changes with grid spacing and the energy changes accordingly.  In the limit the discretized distributions become delta functions, this self energy becomes infinite.

The second calculation you describe is a solvation energy calculation:  it removes the self-energies by performing calculations with the same discretized charges (e.g., same grid spacing and position) but different coefficient values.  The charge-charge interactions can be added back into this energy calculation through Coulomb's law.  See the [APBS examples](https://apbs.readthedocs.io/en/latest/using/examples/solvation-energies.html) for more information.

Thank you,

Nathan
